### PR TITLE
[fb-package] Usability improvements + fix to hesitant_and_* vars

### PR DESCRIPTION
**Pull request by nmdefries**
### Description
General usability improvements to contingency pipeline. Decrease memory usage. Improve clarity and conciseness. Fix some variable definitions (will need to reissue Box tables).

### Changelog
- Initially filter data to only keep columns used in aggregations.
- Filter list of files from `params$input` to leave out unnecessary files, i.e. if a file doesn't contain any responses within the requested aggregation date range. Assumes standard CSV name format.
- Introduce new params setting `use_input_asis` to suppress above feature. Necessary if inputting files with non-standard names or if want to load all data files regardless of specified start and end dates.
- Simplify code that handles removing aggregations when necessary columns are missing.
-  Fix definitions of `hesitant_and_*` variables -- previously denominator was all responses to V3 (vaccine acceptance item), but the intent was to do the calculation *among* hesitant responses only. (This change has already been made in the report data and other data sent to FB.)
- Add default weekly aggs (same as default monthly aggs for now).

### CODE REVIEW

**Comment by krivard** [comment]
looks like you ran into a conflicting change in contingency_variables.R

File: facebook/delphiFacebook/R/contingency_aggregate.R [hunk 1]
```diff
@@ -158,37 +167,43 @@ post_process_aggs <- function(df, aggregations, cw_list) {
...
-    if ( is.null(df[[col_var]]) ) {
-      aggregations <- aggregations[aggregations$metric != col_var &
-                                     !mapply(aggregations$group_by,
-                                             FUN=function(x) {col_var %in% x}), ]
-      msg_plain(
-        paste0(
-          col_var, " is not defined. Removing all aggregations that use it. ", 
-          nrow(aggregations), " remaining")
-      )
+  group_cols <- unique(do.call(c, aggregations$group_by))
```
Comment (capnrefsmmat) [inline facebook/delphiFacebook/R/contingency_aggregate.R:172]: hmm, why `do.call`? Do you want `c(aggregations$group_by, recursive = TRUE)`?

Comment (capnrefsmmat) [reply facebook/delphiFacebook/R/contingency_aggregate.R:172]: and how does this variable differ from `group_vars`?

**Review by capnrefsmmat — COMMENTED** [review]
just one question; otherwise I think this looks fine

**Review by capnrefsmmat — APPROVED** [review]

**Pull request merged**